### PR TITLE
Creation of entity type validation

### PR DESCRIPTION
**Pull request by Mun1z**
Provides validation of entity types on either the API or the Command Line.
Valid Entities: dataset, labels, models, project and repository

Example:

**init** - This command will start the ml-git entity.
```
  api.init('invalid_entity')
```

 **RuntimeError**: The entity type informed is invalid.

### CODE REVIEW

File: ml_git/repository.py [hunk 1]
```diff
@@ -168,6 +171,16 @@ def add(self, spec, file_path, bump_version=False, run_fsck=False):
         if run_fsck:
             self.fsck()
 
+    def _valid_entity_type(self, repo_type):
+
+        type_list = EntityType.list()
+        type_list.append('repository')
+        type_list.append('project')
```
Comment (gilMars) [inline ml_git/repository.py:178]: Add this value in EntityType enum, if possible.

File: ml_git/repository.py [hunk 2]
```diff
@@ -168,6 +171,16 @@ def add(self, spec, file_path, bump_version=False, run_fsck=False):
         if run_fsck:
             self.fsck()
 
+    def _valid_entity_type(self, repo_type):
+
+        type_list = EntityType.list()
```
Comment (gilMars) [inline ml_git/repository.py:176]: Delete the method and change to list comprehension, Ex:
type_list = [entity.value for entity in EntityType]

File: ml_git/utils.py [hunk 3]
```diff
@@ -189,6 +190,14 @@ def change_mask_for_routine(is_shared_path=False):
         yield
 
 
+@contextmanager
+def disable_exception_traceback():
+    default_value = getattr(sys, "tracebacklimit", 1000)
```
Comment (gilMars) [inline ml_git/utils.py:195]: Change double quote to single quote.

File: ml_git/repository.py (hunk 2 above)
Comment (lucasncabral) [reply ml_git/repository.py:176]: I think we can keep the method that lists the values ​​of the enum, but change the way those values ​​are listed to the way that @gilMars commented.

File: tests/integration/test_24_api.py [hunk 4]
```diff
@@ -330,3 +330,25 @@ def test_16_create_entity_with_import(self):
...
+    def test_17_create_with_invalid_entity(self):
+        try:
+            entity_type = 'dataset_invalid'
+            store_type = StoreType.S3H.value
+            self.assertIn(output_messages['INFO_INITIALIZED_PROJECT'] % self.tmp_dir, check_output(MLGIT_INIT))
+            api.create('dataset_invalid', 'dataset-ex', categories=['computer-vision', 'images'], mutability='strict')
+            self.check_created_folders(entity_type, store_type)
+            self.assertTrue(False)
+        except Exception:
+            self.assertTrue(True)
```
Comment (lucasncabral) [inline tests/integration/test_24_api.py:344]: Add the validation that the exception being received is the one that expected: 'The entity type informed is invalid.'

File: ml_git/utils.py (hunk 3 above)
Comment (Mun1z) [reply ml_git/utils.py]: Done on commit [4a6ab83](https://github.com/HPInc/ml-git/pull/56/commits/4a6ab8372f84c38c2d0a82b125e4351cc92da114)

File: tests/integration/test_24_api.py (hunk 4 above)
Comment (Mun1z) [reply tests/integration/test_24_api.py]: Done on commit [4a6ab83](https://github.com/HPInc/ml-git/pull/56/commits/4a6ab8372f84c38c2d0a82b125e4351cc92da114)

File: ml_git/repository.py (hunk 2 above)
Comment (Mun1z) [reply ml_git/repository.py]: Done on commit [4a6ab83](https://github.com/HPInc/ml-git/pull/56/commits/4a6ab8372f84c38c2d0a82b125e4351cc92da114)

**Review by lucasncabral — APPROVED** [review]

File: ml_git/repository.py (hunk 1 above)
Comment (Mun1z) [reply ml_git/repository.py:176]: This change will not be possible, as it ends up breaking the update_entities_ metadata method on repository.py

Comment (andersonpablosilva) [reply ml_git/repository.py:176]: I believe that we should not add to EntityType, since they are not ML entity types.

**Review by andersonpablosilva — APPROVED** [review]

**Review by gilMars — APPROVED** [review]

File: ml_git/repository.py [hunk 5]
```diff
@@ -169,6 +171,14 @@ def add(self, spec, file_path, bump_version=False, run_fsck=False):
         if run_fsck:
             self.fsck()
 
+    def _valid_entity_type(self, repo_type):
```
Comment (diego-pedroso) [inline ml_git/repository.py:174]: What about `_validate_entity_type`?

Comment (Mun1z) [reply ml_git/repository.py]: Done on commit [deb72ee](https://github.com/HPInc/ml-git/pull/56/commits/deb72ee18ace614ff800d6fd2cdd17c9f5b436fe)

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**